lib/Factory.php

```diff
@@ -21,7 +21,7 @@ class Factory
     public const LATEX = 'LaTeX';
 
     /** @var string */
-    private $name;
+    protected $name;
```

Comment:
I don't think this is the correct approach for extensibility, especially since this class is a lot of switching based on a constructor argument. Instead, adding an interface for the getters to allow consumers to provide their own  (arbitrarily complex) instantiation logic, as well as their own dependencies via constructor arguments, would be preferable.

Reply:
Probably worth adding a note (at class level docblock) that this class is not meant to be extended

Reply:
They can always wrap the original factory

Reply:
Oh, right. Good call :+1: